src/components/Map/MapRenderer/MapRenderer.js

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { GoogleMap, Marker, Circle } from '@react-google-maps/api';
+
+const circleOptions = radius => ({
+    strokeColor: 'transparent',
+    strokeOpacity: 1,
+    strokeWeight: 0,
+    fillColor: '#FF0000',
+    fillOpacity: 0.35,
+    clickable: false,
+    draggable: false,
+    editable: false,
+    visible: true,
+    radius,
+    zIndex: 1
+});
+
+const MapRenderer = React.forwardRef((props, ref) => {
+    const { center, zoom, markers, mapContainerStyle, ...rest } = props;
+    const positions = markers.map(marker => ({ lat: marker.lat, lng: marker.lng }));
+    const mapRef = ref || React.createRef();
+
+    const fitBounds = () => {
```

Comment:
Needs to be memoized with `useCallback`

Reply:
ok. there are a few things to polish still...

Reply:
still missing `useCallback `